lib/plugins/aws/provider/awsProvider.js

```diff
@@ -482,6 +482,34 @@ class AwsProvider {
     return { Ref: this.naming.getRestApiLogicalId() };
   }
 
+  /**
+   * Get API Gateway Rest API Stage description.
+   * Keys are expected in camelCase and will be exported in cloudformations PascalCase
+   *
+   * See https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-apigateway-deployment-stagedescription.html
+   * for informatation on cloud formation support of StageDescriptions
+   */
+  getApiGatewayStageDescription() {
+    if (this.serverless.service.provider.apiGateway
+      && this.serverless.service.provider.apiGateway.stageDescription) {
+      const stageDescription = this.serverless.service.provider.apiGateway.stageDescription;
+      if (!_.isPlainObject(stageDescription)) {
+        throw new Error('REST API stage description must be an object');
+      }
+      const tracingEnabled = stageDescription.tracingEnabled;
+      if (tracingEnabled !== undefined) {
+        if (!_.isBoolean(tracingEnabled)) {
+          throw new Error('REST API stage description tracingEnabled must be a boolean');
+        }
+        return {
```

Comment:
I'm letting `tracingEnabled` be the only supported (and documented) field associated with stage descriptions for now, but having this method in place should make it straightforward and easy to add additional supported fields in the future. Something I've not yet figured out is is there is a better place to communicate structural validation prior to `serverless deploy` invocations.